codegen/core/src/main/scala/com/lightbend/akkasls/codegen/SourceGeneratorUtils.scala

```diff
@@ -43,13 +43,29 @@ object SourceGeneratorUtils {
     """// This code is managed by Akka Serverless tooling.
        |// It will be re-generated to reflect any changes to your protobuf definitions.
        |// DO NOT EDIT""".stripMargin
-
+  //TODO delete this val
   val unmanagedComment =
     """// This class was initially generated based on the .proto definition by Akka Serverless tooling.
        |//
        |// As long as this file exists it will not be overwritten: you can maintain it yourself,
        |// or delete it so it is regenerated as needed.""".stripMargin
 
+  def unmanagedComment(service: Either[ModelBuilder.Service, ModelBuilder.Entity]) = {
+
+    val (kind, fileName) = service match {
+      case Left(serv: ModelBuilder.ActionService)      => ("Action Service", serv.fqn.parent.protoFileName)
+      case Left(serv: ModelBuilder.ViewService)        => ("View Service", serv.fqn.parent.protoFileName)
+      case Right(ent: ModelBuilder.EventSourcedEntity) => ("Event Sourced Entity Service", ent.fqn.parent.protoFileName)
+      case Right(ent: ModelBuilder.ValueEntity)        => ("Value Entity Service", ent.fqn.parent.protoFileName)
+      case Right(ent: ModelBuilder.ReplicatedEntity)   => ("Replicated Entity Service", ent.fqn.parent.protoFileName)
+    }
+    s"""// This class was initially generated based on the .proto definition by Akka Serverless tooling.
+       |// This is the implementation for the $kind described in your $fileName file.
```

Comment:
include the relative proto path if possible, I guess we don't know if a user remapped `src/main/proto` in their project, but for `src/main/proto/com/example/my_service.proto` getting a `com/example/my_service.proto` would be nice

Reply:
I tried and didn't see the path in BuilderModel.Service or Entity. I'll have another look around to see what can use.

Reply:
aren't you dropping the parent path just above when you pick up only the `protoFileName` from the fully qualified name? 

Perhaps @octonato knows from the top of his head given that he has been mucking about with the modelbuilder lately.

Reply:
It's all good. Thanks @johanandren